app/config/ErrorHandler.scala

```diff
@@ -17,16 +17,42 @@
 package config
 
 import javax.inject.{Inject, Singleton}
+import connectors.httpparsers.IncomeSourcesHttpParser.{IncomeSourcesError, IncomeSourcesServiceUnavailableError}
 import play.api.i18n.MessagesApi
-import play.api.mvc.Request
+import play.api.mvc.Results._
+import play.api.mvc.{Request, RequestHeader, Result}
 import play.twirl.api.Html
 import uk.gov.hmrc.play.bootstrap.frontend.http.FrontendErrorHandler
+import views.html.errors.{InternalServerErrorPage, NotFoundPage, ServiceUnavailablePage}
 import views.html.templates.ErrorTemplate
 
+import scala.concurrent.Future
 @Singleton
-class ErrorHandler @Inject()(errorTemplate: ErrorTemplate, val messagesApi: MessagesApi)(implicit appConfig: AppConfig)
+class ErrorHandler @Inject()(errorTemplate: ErrorTemplate, val messagesApi: MessagesApi,
+                             internalServerErrorPage: InternalServerErrorPage, notFoundPage: NotFoundPage,
+                             serviceUnavailablePage: ServiceUnavailablePage)(implicit appConfig: AppConfig)
+
   extends FrontendErrorHandler {
 
   override def standardErrorTemplate(pageTitle: String, heading: String, message: String)(implicit request: Request[_]): Html =
-    errorTemplate(pageTitle, heading, message)
-}
\ No newline at end of file
+    internalServerErrorPage()
+
+  override def notFoundTemplate(implicit request: Request[_]): Html = notFoundPage()
+
+  override def internalServerErrorTemplate(implicit request: Request[_]): Html = internalServerErrorPage()
+
+  def handleError(error: IncomeSourcesError)(implicit request: Request[_]): Result = {
+    error match {
+      case IncomeSourcesServiceUnavailableError => ServiceUnavailable(serviceUnavailablePage())
+      case _ => InternalServerError(internalServerErrorPage())
+    }
+  }
+
+  override def onClientError(request: RequestHeader, statusCode: Int, message: String): Future[Result] =
+    statusCode match {
+      case play.mvc.Http.Status.NOT_FOUND =>
+        Future.successful(NotFound(notFoundTemplate(request.withBody(""))))
+      case _ =>
+        Future.successful(BadRequest(badRequestTemplate(request.withBody(""))))
```

Comment:
sorry forgot about this one but could you make the case _ InternalServerError(internalServerErrorPage())